### PR TITLE
maintainers: switch from list to directive

### PR DESCRIPTION
**Pull request by tldahlgren**
Since it is not unusual for PRs to be submitted using the old `maintainers` syntax (i.e., a list), this PR replaces all but one occurrence in the repo with the new syntax (i.e., the directive described at https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers).  I also ran `black` on several unit test data files to fix single-to-double quotes and indentation.

It should be noted that a list is still a valid way to override (instead of add to via the directive) any inherited maintainers.  Hence there is one mock package (`py-extension2`) that retains the list since it is a deliberate example of the override capability.

If any of the modified packages are deliberately meant to override the maintainers with a list, then I will restore the syntax.  Just let me know.

### CODE REVIEW

**Review by wdconinc — APPROVED** [review]
Approving dpmjet, estarlight, py-minkowskiengine.

**Review by adamjstewart — APPROVED** [review]

**Review by ulmononian — APPROVED** [review]
looks A-OK.

**Review by tom91136 — COMMENTED** [review]
LGTM!

**Review by srekolam — COMMENTED** [review]
LGTM!!

**Comment by dmay23** [comment]
Looks good.

**Comment by mdorier** [comment]
Looks good to me.

**Review by Thomas-Ulrich — COMMENTED** [review]
LGTM

**Comment by wrwilliams** [comment]
Fine for scorep.

File: var/spack/repos/builtin/packages/scorep/package.py
Comment (wrwilliams) [inline var/spack/repos/builtin/packages/scorep/package.py:1]: LGTM

**Pull request merged**